### PR TITLE
CONSVC 1813: Include iOS data into Contextual Services derived dataset

### PR DESCRIPTION
**Pull request by ncloudioj**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

r? @jklukas @whd 

This fixes [CONSVC-1813](https://mozilla-hub.atlassian.net/browse/CONSVC-1813).

Since this change is only about Top Sites, also with the recent policy change,  does it still requires a CCP?

### CODE REVIEW

**Comment by jklukas** [comment]
> Since this change is only about Top Sites, also with the recent policy change, does it still requires a CCP?

No, I agree that this doesn't require CCP since it's unrelated to search terms data. And the affected files here aren't in the codeowners paths, so SRE review isn't technically enforced either here.

**Review by jklukas — APPROVED** [review]
These changes look good. I appreciate the numerous code comments to explain the constants and the `IFNULL` statements.

I pushed to upstream to rerun CI with credentials. If that all passes, this should be good to merge.

File: sql/moz-fx-data-shared-prod/contextual_services_derived/event_aggregates_v1/query.sql [hunk 1]
```diff
@@ -172,20 +159,68 @@ WITH combined AS (
...
+    'topsites' AS source,
+    IF(events[SAFE_OFFSET(0)].name = 'contile_click', 'click', 'impression') AS event_type,
+    -- The adMarketplace APIs accept form factors out of "desktop", "phone", or "tablet";
+    -- we are currently always using "phone" for Fenix, so stay consistent with that here.
+    'phone' AS form_factor,
+    normalized_country_code AS country,
+    metadata.geo.subdivision1 AS subdivision1,
+    IFNULL(metrics.string.top_sites_contile_advertiser, metrics.string.top_site_contile_advertiser) AS advertiser,
+    'release' AS release_channel,
+    CAST(
```
Comment (quiiver) [inline sql/moz-fx-data-shared-prod/contextual_services_derived/event_aggregates_v1/query.sql:185]: do the `CAST` calls in here need to be `SAFE_CAST` to guard against non int values?

**Review by quiiver — APPROVED** [review]
Have a small question about the `CAST` calls but otherwise this looks good to me.

File: sql/moz-fx-data-shared-prod/contextual_services_derived/event_aggregates_v1/query.sql (hunk 1 above)
Comment (jklukas) [reply sql/moz-fx-data-shared-prod/contextual_services_derived/event_aggregates_v1/query.sql:185]: I'd agree it's generally a good idea to prefer `SAFE_CAST`

**Comment by ncloudioj** [comment]
@jklukas @quiiver Thank you both for the review!

And all the tests passed locally, whew.

**Pull request merged**